### PR TITLE
PML/SPML/UCX: Adapt to the API changes in the UCX lib.

### PR DESCRIPTION
**Pull request by alinask**
Signed-off-by: Alina Sklarevich alinas@mellanox.com

### CODE REVIEW

**Comment by alinask** [comment]
@yosefe @alex-mikheev @jladd-mlnx please review

**Comment by yosefe** [comment]
:+1:  looks good

**Comment by alinask** [comment]
This PR isn't ready for merging since the UCX API change isn't present in UCX repo yet.

**Comment by jsquyres** [comment]
@alinask I added the "WIP-DNM" label (work in progress / do not merge).  Feel free to remove that label when it is ready to be merged.

Please also add the appropriate signed-off-by token in the commit message (notice that that CI test failed).  Thanks.

File: ompi/mca/pml/ucx/pml_ucx.c [hunk 1]
```diff
@@ -175,13 +175,17 @@ int mca_pml_ucx_close(void)
...
+    ucp_worker_params_t params;
     ucs_status_t status;
     int rc;
 
     PML_UCX_VERBOSE(1, "mca_pml_ucx_init");
 
     /* TODO check MPI thread mode */
-    status = ucp_worker_create(ompi_pml_ucx.ucp_context, UCS_THREAD_MODE_SINGLE,
+    params.field_mask  = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    params.thread_mode = UCS_THREAD_MODE_SINGLE;
```
Comment (miked-mellanox) [inline ompi/mca/pml/ucx/pml_ucx.c]: @alinask - shall we auto-detect it based on MPI_Init or MPI_Init_thread was called?

**Comment by alinask** [comment]
@jsquyres I updated the commit message, thanks!
I see that the only Jenkins that failed was the Mellanox one - this is because the UCX library doesn't have the API changes yet.

**Comment by alinask** [comment]
@miked-mellanox The thread mode selection will be added in a later PR. 
This PR is only for the UCX API changes.

**Comment by jsquyres** [comment]
@alinask I was referring to the "Signed-off-by" checker CI test; it also failed.  It looks like the commit message here on Github still doesn't have the signed off by token -- perhaps you didn't push it yet?  Be aware that if you `git commit --amend -s` to add the singed-off-by token, you'll need to `git push YOUR_REMOTE --force` to force push to get that commit up here to Github (because you're changing git history).

Note that just about the *only* time it is acceptable to force push is for pull requests that have not yet been merged.  😄 

But as a second point: am I correct in thinking that UCX has not been released yet?  I.e., it's ok to make changes to how the UCX API is used in OMPI without adding corresponding configury changes to check and see what version of the UCX API is being used?

**Comment by hppritcha** [comment]
Will these changes also target 2.1.x branch?

**Comment by alinask** [comment]
@jsquyres I added the 'Signed-off-by' token, thanks!

@hppritcha  Yes, they will.

**Pull request merged**